pyodide_build/pypabuild.py

```diff
@@ -153,26 +159,29 @@ def _build_in_isolated_env(
 
         # first install the build dependencies
         symlink_unisolated_packages(env)
-        install_reqs(env, builder.build_system_requires)
-        installed_requires_for_build = False
+        install_reqs(build_env, env, builder.build_system_requires)
+        build_reqs = None
         try:
             build_reqs = builder.get_requires_for_build(
                 distribution,
             )
         except BuildBackendException:
             pass
-        else:
-            install_reqs(env, build_reqs)
-            installed_requires_for_build = True
 
-        with common.replace_env(build_env):
-            if not installed_requires_for_build:
+        if not build_reqs:
+            # get_requires_for_build in native env failed. Maybe trying to
+            # execute get_requires_for_build in the cross build environment will
+            # work?
+            # TODO: Is this case ever used? Add test coverage or remove.
```

Comment:
I saw that this case was used in pygame-ce. In native env, the setup.py picks up native SDL2 config, then fails. In the cross env, it correctly picks up Emscripten SDL2 config.

[pygame-ce.log](https://github.com/user-attachments/files/19269447/pygame-ce.log)

Reply:
Updated the comment.

Reply:
I guess a second question is whether the first attempt is ever necessary. It would be nice to only have one of these. But we can investigate later.

Reply:
Well, packages that are unaware of Emscripten would have something like:

```python
match sys.platform:
  case 'linux': do_somthing_linux()
  case 'macos': do_somthing_macos()
  case 'windows': dos_something_windows()
  default: raise RuntimeError("now supported platform")
```

and in this case, we would like to choose linux for a best effort.

Reply:
In that case, maybe we could move towards selecting which of these behaviors is desired in the `meta.yaml` file. It would be nice to be more explicit about this. But perhaps no action is needed.